crates/subspace-rpc-primitives/src/lib.rs

```diff
@@ -87,3 +87,20 @@ pub struct RewardSignatureResponse {
     /// Pre-header or vote hash signature.
     pub signature: Option<RewardSignature>,
 }
+
+/// Information about new slot that just arrived
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum NodeSyncStatus {
+    /// Node is fully synced
+    Synced,
+    /// Node is major syncing
+    MajorSyncing,
```

Comment:
What does `Major` mean here?

Reply:
It is Susbtrate's concept. Major syncing means local chain is far behind and syncing, no blocks are produced in the meantime. If local chain is a few blocks behind, it'll still pull those blocks and sync, but it is not a major synching in that case and blocks authoring can still happen. I have called it `MajorSyncing` like in Substrate with `fn is_synced` in case we need to extend enum with more detailed status in the future (but we don't need it right now and may never need it).